sql/OverdueAnalysis.sql

```diff
@@ -0,0 +1,7 @@
+-- List all books overdue by more than 30 days with their associated borrowers.
+SELECT Borrowers.FirstName, Borrowers.LastName, Books.Title, Loans.DateBorrowed, Loans.DueDate
+FROM Loans
+         JOIN Borrowers ON Loans.BorrowerID = Borrowers.BorrowerID
+         JOIN Books ON Loans.BookID = Books.BookID
+WHERE DATEDIFF(DAY, Loans.DueDate, GETDATE()) > 30
```

Comment:
this means that the datediff function will be called on each record to filter data. think of a way to avoid this. and read on using functions in where conditions.

Reply:
DECLARE @ThirtyDaysAgo DATE = DATEADD(DAY, -30, GETDATE());
```suggestion
DECLARE @ThirtyDaysAgo DATE = DATEADD(DAY, -30, GETDATE());
```

This variable avoids recalculating "30 days ago" multiple times. it is calculated once, and stored it in @ThirtyDaysAgo